### PR TITLE
fix: fix decompose_generators_fold (PROOF-620)

### PR DESCRIPTION
**Pull request by rnburn**
# Rationale for this change

Fix bug that appears with full optimizations. I was using the incorrect value when doing a bit shift.

# What changes are included in this PR?

A fix for bug.

# Are these changes tested?

Functionality is covered under existing tests. I'm going to follow up with a PR that runs tests with full optimizations in the CI.

### CODE REVIEW

File: sxt/proof/inner_product/generator_fold.cc [hunk 1]
```diff
@@ -33,6 +33,11 @@ void decompose_generator_fold(basct::span<unsigned>& res, const s25t::element& m
                               const s25t::element& m_high) noexcept {
   SXT_DEBUG_ASSERT(res.size() == s25cn::max_bits_v);
   size_t bit_index = 0;
+
+  // workaround for clang bug
+  // see https://github.com/llvm/llvm-project/issues/66568
+  volatile auto data = res.data();
```
Comment (joe-stifler) [inline sxt/proof/inner_product/generator_fold.cc:39]: Well done, Ryan! 🎉 🎉

**Review by joe-stifler — DISMISSED** [review]
Thank you, Ryan! I've successfully built the library from this pull request, and it's functioning perfectly on my local setup.

File: sxt/proof/inner_product/generator_fold.cc (hunk 1 above)
Comment (rnburn) [reply sxt/proof/inner_product/generator_fold.cc]: Actually, looks like it was a bug on my part. But I put in a fix.

Comment (joe-stifler) [reply sxt/proof/inner_product/generator_fold.cc]: Still, it was pretty cool that you managed to spot the implementation.

**Review by joe-stifler — APPROVED** [review]

**Pull request merged**